minigotchi-ESP32/display.cpp

```diff
@@ -3,58 +3,97 @@
+      tft.begin();  // Initialize TFT_eSPI library
+      tft.setRotation(1);  // Set display rotation if needed
+      tft.fillScreen(TFT_BLACK);  // Fill screen with black color
+      tft.setTextColor(TFT_WHITE);  // Set text color to white
+      tft.setTextSize(2);  // Set text size
     } else {
       // use wemos shield by default
       adafruit_display = new Adafruit_SSD1306(WEMOS_OLED_SHIELD_OLED_RESET);
-      adafruit_display->begin(SSD1306_SWITCHCAPVCC, 0x3C); // initialize with the I2C addr 0x3C (for the 64x48)
+      if (!adafruit_display->begin(SSD1306_SWITCHCAPVCC, 0x3C)) { // initialize with the I2C addr 0x3C
+        Serial.println(F("SSD1306 allocation failed"));
+        return;
+      }
+      delay(100); // initialize w/ delays to prevent crash
+      adafruit_display->display();
+      delay(100);
+      adafruit_display->clearDisplay();
+      adafruit_display->setTextColor(WHITE);
     }
-
-    delay(5);
-    // initialize w/ delays to prevent crash
-    adafruit_display->display();
-    delay(5);
-    adafruit_display->clearDisplay();
-    delay(5);
-    adafruit_display->setTextColor(WHITE);
-    delay(5);
   }
 }
 
 void Display::cleanDisplayFace(String text) {
   if (Config::display) {
-    adafruit_display->setCursor(0, 0);
-    delay(5);
-    adafruit_display->setTextSize(2);
-    delay(5);
-    adafruit_display->clearDisplay();
-    delay(5);
-    adafruit_display->println(text);
-    delay(5);
-    adafruit_display->display();
-    delay(5);
+    if (Config::screen == "CYD") {   // Check if the screen configuration is set to "CYD" and execute the corresponding code
+      tft.fillScreen(TFT_BLACK);  // Clear screen with black color
+      tft.setTextColor(TFT_WHITE);  // Set text color to WHITE
+      tft.setCursor(0, 5);  // Set cursor position
+      tft.setTextSize(4);  // Set text size
+      tft.println(text);  // Print text
+    } else {
+      adafruit_display->clearDisplay();
+      adafruit_display->setTextSize(2);
+      adafruit_display->setCursor(0, 0);
+      adafruit_display->println(text);
+      adafruit_display->display();
+    }
   }
 }
 
 void Display::attachSmallText(String text) {
   if (Config::display) {
-    adafruit_display->setCursor(0, 20);
-    delay(5);
-    adafruit_display->setTextSize(1);
-    delay(5);
-    adafruit_display->println(text);
-    delay(5);
-    adafruit_display->display();
-    delay(5);
+    if (Config::screen == "CYD") {   // Check if the screen configuration is set to "CYD" and execute the corresponding code
+      tft.setTextColor(TFT_WHITE);  // Set text color to WHITE
+      tft.setCursor(0, 40);  // Set cursor position
+      tft.setTextSize(1);  // Set text size
+      tft.println(text);  // Print text
+    } else {
+      adafruit_display->setCursor(0, 20);
+      adafruit_display->setTextSize(1);
+      adafruit_display->println(text);
+      adafruit_display->display();
+    }
   }
-}
\ No newline at end of file
+}
```

Comment:
is there any issue regarding the screen libraries? are they clashing at all or were your changes for mere improvement? as of right now it seems like the (SSD1306/WEMOS shield) screens work just fine.